action.yml

```diff
@@ -67,57 +67,57 @@ runs:
         if [[ -n "${{ inputs.config }}" ]]
         then
             echo "using config from '${{ inputs.config }}'"
-            CONFIG_PATH="--config ${{ inputs.config }}"
+            CONFIG_PATH=("--config" "${{ inputs.config }}")
         else
-            CONFIG_PATH=""
+            CONFIG_PATH=()
         fi
 
         if [[ -n "${{ inputs.token }}" ]]
         then
             echo "using custom registry token'"
-            TOKEN="--token ${{ inputs.token }}"
+            TOKEN=("--token" "${{ inputs.token }}")
         else
-            TOKEN=""
+            TOKEN=()
         fi
 
         if [[ -n "${{ inputs.backend }}" ]]
         then
           echo "using backend '${{ inputs.backend }}'"
-          BACKEND="--backend ${{ inputs.backend }}"
+          BACKEND=("--backend" "${{ inputs.backend }}")
         else
-          BACKEND=""
+          BACKEND=()
         fi
 
         if [[ -n "${{ inputs.registry }}" ]]
         then
             echo "using registry '${{ inputs.registry }}'"
-            ALT_REGISTRY="--registry ${{ inputs.registry }}"
+            ALT_REGISTRY=("--registry" "${{ inputs.registry }}")
         else
-            ALT_REGISTRY=""
+            ALT_REGISTRY=()
         fi
 
         if [[ -n "${{ inputs.manifest_path }}" ]]
         then
             echo "using manifest path '${{ inputs.manifest_path }}'"
-            MANIFEST_PATH="--manifest-path ${{ inputs.manifest_path }}"
+            MANIFEST_PATH=("--manifest-path "${{ inputs.manifest_path }}")
         elif [[ -n "${{ inputs.project_manifest }}" ]]
         then
             echo "using manifest path '${{ inputs.project_manifest }}'"
-            MANIFEST_PATH="--project-manifest ${{ inputs.project_manifest }}"
+            MANIFEST_PATH=("--project-manifest" "${{ inputs.project_manifest }}")
         else
-            MANIFEST_PATH=""
+            MANIFEST_PATH=()
         fi
 
         if [[ -z "${{ inputs.command }}" || "${{ inputs.command }}" == "release-pr" ]]
         then
             echo "-- Running release-plz release-pr --"
             release_pr_output=$(release-plz release-pr\
-                --git-token ${GITHUB_TOKEN}\
-                --repo-url https://github.com/${GITHUB_REPOSITORY}\
-                ${CONFIG_PATH}\
-                ${ALT_REGISTRY}\
-                ${MANIFEST_PATH}\
-                ${BACKEND}\
+                --git-token "${GITHUB_TOKEN}"\
+                --repo-url "https://github.com/${GITHUB_REPOSITORY}"\
+                "${CONFIG_PATH[@]}"\
+                "${ALT_REGISTRY[@]}"\
+                "${MANIFEST_PATH[@]}"\
+                "${BACKEND[@]}"\
```

Comment:
is this trick used to work with whitespace? If yes, why are we using it also in the backend, which is one world only?
Same for ALT_REGISTRY probably

Reply:
yes this trick is to keep the separation of argument clean as defined in the array. I also apply it BACKEND and ALT_REGISTRY for:
- consistency (same pattern for every arguments fragment)
- to guide if new argument should be added in the future (no question about which "form" to use)